src/main/java/org/skriptlang/skript/lang/script/ScriptWarning.java

```diff
@@ -18,15 +18,49 @@
  */
 package org.skriptlang.skript.lang.script;
 
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.parser.ParserInstance;
+import ch.njol.util.Kleenean;
+
 /**
  * An enum containing {@link Script} warnings that can be suppressed.
  */
 public enum ScriptWarning {
 
-	VARIABLE_SAVE, // Variable cannot be saved (the ClassInfo is not serializable)
+	/**
+	 * Variable cannot be saved (the ClassInfo is not serializable)
+	 */
+	VARIABLE_SAVE,
+
+	/**
+	 * Missing "and" or "or"
+	 */
+	MISSING_CONJUNCTION,
+
+	/**
+	 * Variable starts with an Expression
+	 */
+	VARIABLE_STARTS_WITH_EXPRESSION,
 
-	MISSING_CONJUNCTION, // Missing "and" or "or"
+	/**
+	 * This syntax is deprecated and scheduled for future removal
+	 */
+	DEPRECATED_SYNTAX;
 
-	VARIABLE_STARTS_WITH_EXPRESSION // Variable starts with an Expression
+	/**
+	 * Prints the given message using {@link Skript#warning(String)} iff the current script does not suppress deprecation warnings.
+	 * Intended for use in {@link ch.njol.skript.lang.SyntaxElement#init(Expression[], int, Kleenean, SkriptParser.ParseResult)}.
+	 *
+	 * @param message the warning message to print.
+	 */
+	public static void printDeprecationWarning(String message) {
+		ParserInstance parser = ParserInstance.get();
+		Script currentScript = parser.isActive() ? parser.getCurrentScript() : null;
+		if (currentScript != null && currentScript.suppressesWarning(ScriptWarning.DEPRECATED_SYNTAX))
+			return;
+		Skript.warning(message);
```

Comment:
Can we add a spceial prefix? like `[Deprecated]`

Reply:
Good idea, should help people realize they can suppress it